awx/main/tests/unit/test_tasks.py

```diff
@@ -1053,7 +1068,7 @@ def test_gce_credentials(self, private_data_dir, job):
         safe_env = {}
         credential.credential_type.inject_credential(credential, env, safe_env, [], private_data_dir)
         runner_path = env['GCE_CREDENTIALS_FILE_PATH']
-        local_path = os.path.join(private_data_dir, os.path.basename(runner_path))
+        local_path = runner_path.replace('/runner', private_data_dir)  # container path to host path
```

Comment:
I know these are only the tests, but this looks like the direction I would like to start taking things in.

There _might_ be corner cases that this will have problems with... but in the big picture, this is a durable way to switch from the container to host path.